crypto/merkle/simple_tree.go

```diff
@@ -51,6 +56,7 @@ func simpleHashFromHashes(hashes [][]byte) []byte {
 	case 1:
 		return hashes[0]
 	default:
+		// TODO: Change how we split the left and right nodes
```

Comment:
Wasn't this the part @jaekwon was sceptical about? 

According to the [rfc][1] this would change to:

(n is the ordered number of inputs d(0), ..., d(n-1) )

> The hash of a list with one entry (also known as a leaf hash) is:
> MTH({d(0)}) = SHA-256(0x00 || d(0)).

>    For n > 1, let k be the largest power of two smaller than n (i.e.,
>   k < n <= 2k).  The Merkle Tree Hash of an n-element list D[n] is then
>   defined recursively as
>
>   MTH(D[n]) = SHA-256(0x01 || MTH(D[0:k]) || MTH(D[k:n])),
>
 >  where || is concatenation and D[k1:k2] denotes the list {d(k1),
 >  d(k1+1),..., d(k2-1)} of length (k2 - k1).  (Note that the hash
 >  calculations for leaves and nodes differ.  This domain separation is
 >  required to give second preimage resistance.)

This shouldn't be difficult to change, right?

[1]: https://tools.ietf.org/html/rfc6962#section-2.1